### PR TITLE
Remove `BuiltInWindowFunction` (LogicalPlans)

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

- (Almost closes) https://github.com/apache/datafusion/issues/8709 :tada:!!!
- Follow on https://github.com/apache/datafusion/pull/13201

## Rationale for this change

@buraksen and @jcsherin ported the last window function over, so now we can remove the old BuiltInWindowFunction code

This was so tempting a cleanup that I had to give it a try (and I wanted to do some coding)

## What changes are included in this PR?

Changes:
1. Remove `BuiltInWindowFunction`
2. Remove `BuiltInWindowFunctionDefinition::BuiltInWindowFunction`
3. Remove Protobuf definitions
5. Update code to compile

## Are these changes tested?
Yes, by existing CI

### CODE REVIEW

**Comment by alamb** [comment]
There are actually several related "BuiltInWindowExpr" in physical-expr that need to be cleaned up as well (that are no longer related to BuiltInWindowExprs). We may also be able to remove a level of trait.

**Comment by jonathanc-n** [comment]
@alamb @buraksenn @Omega359 Pretty sure we can now clean up the old window docs, due to the nth_value migration being merged. Lemme see if I can do that really quick

**Comment by alamb** [comment]
> @alamb @buraksenn @Omega359 Pretty sure we can now clean up the old window docs, due to the nth_value migration being merged. Lemme see if I can do that really quick

Yes, we can -- I think that is tracked by
- https://github.com/apache/datafusion/issues/13171

File: datafusion/expr/src/expr.rs [hunk 1]
```diff
@@ -697,13 +694,13 @@ impl AggregateFunction {
     }
 }
 
-/// WindowFunction
+/// A function used as a SQL window function
+///
+/// In SQL, you can use:
+/// - Actual window functions ([`WindowUDF`])
+/// - Noraml aggregate functions ([`AggregateUDF`])
```
Comment (jcsherin) [inline datafusion/expr/src/expr.rs:701]: typo here: "Normal"

Comment (alamb) [reply datafusion/expr/src/expr.rs]: Fixed in [6b1a243](https://github.com/apache/datafusion/pull/13393/commits/6b1a2439d44464d7f49c0c9f4df4f5f67f172344)

**Comment by alamb** [comment]
Thanks @crepererum  and @jcsherin for the reviews

**Pull request merged**